tests/generatedJSONSchemasToFiles.js

```diff
@@ -82,7 +82,7 @@ collections.forEach(collection => {
     const schema = generator[method]()
     let newExpectedSchema = structuredClone(schema)
     const paths = JSONPath({
-      path: '$..*[?(@.type=="string" && @.pattern)]',
+      path: '$..[?(@.type=="string" && @.pattern)]',
```

Comment:
Just out of curiosity, why do we remove the `*` here?

Reply:
To consider the entire object. With *, we would exclude the root level https://github.com/JSONPath-Plus/JSONPath#:~:text=//*/*%7C//*/*/text(),%24..*